src/include/daos_srv/bio.h

```diff
@@ -764,4 +772,48 @@ int bio_replace_dev(struct bio_xs_context *xs, uuid_t old_dev_id,
  */
 int bio_set_led_state(struct bio_xs_context *xs, uuid_t devid,
 		      const char *led_state, bool reset);
+
+/*
+ * Allocate DMA buffer, the buffer could be from bulk cache if bulk context
+ * if specified.
```

Comment:
s/if specified/is specified/
On the other hand, it seems that "bulk_ctxt" is always specified, because you assert "D_ASSERT(ctxt != NULL && ctxt->bic_umem != NULL);".

Reply:
That assert is for io context (bio_io_context) but not bulk context.